blockonomics-woocommerce.php

```diff
@@ -501,19 +508,30 @@ function blockonomics_activation_hook() {
 }
 
 // Since WP 3.1 the activation function registered with register_activation_hook() is not called when a plugin is updated.
+// blockonomics_update_db_check() is loaded for every PHP page by plugins_loaded hook
 function blockonomics_update_db_check() {
-    global $wpdb;
     global $blockonomics_db_version;
-
     $installed_ver = get_site_option( 'blockonomics_db_version' );
-    if ( $installed_ver != $blockonomics_db_version ) {
+    // blockonomics_create_table() should only be run if there is no $installed_ver, refer https://github.com/blockonomics/woocommerce-plugin/issues/296
+    if (empty($installed_ver)){
+        blockonomics_create_table();
+    } else if (version_compare( $installed_ver, $blockonomics_db_version, '!=')) {
+        blockonomics_run_db_updates($installed_ver);
+    }
+}
+
+function blockonomics_run_db_updates($installed_ver){
+    global $wpdb;
+    global $blockonomics_db_version;
+    if (version_compare($installed_ver, '1.1', '<')){
         $table_name = $wpdb->prefix . 'blockonomics_orders';
-        if ($installed_ver < 1.1) {
-            maybe_drop_column($table_name, "time_remaining", "ALTER TABLE $table_name DROP COLUMN time_remaining");
-            maybe_drop_column($table_name, "timestamp", "ALTER TABLE $table_name DROP COLUMN timestamp");
-        }
+        maybe_drop_column($table_name, "time_remaining", "ALTER TABLE $table_name DROP COLUMN time_remaining");
+        maybe_drop_column($table_name, "timestamp", "ALTER TABLE $table_name DROP COLUMN timestamp");
+    }
```

Comment:
Retaining the upgrade scripts ensures that any updates to the plugin between any versions will function correctly and behave consistently. Retaining the upgrade from the previous version allows users to update from versions <= 3.5.7 to version 3.5.8, rather than the most recent version 3.5.9 if they so choose.